runtime/bcutil/defineclass.c

```diff
@@ -140,6 +146,23 @@ internalDefineClass(
 		/* localBuffer should not be NULL */
 		Trc_BCU_Assert_True(NULL != localBuffer);
 
+		/* Host class can only be set for anonymous classes, which are defined by Unsafe.defineAnonymousClass.
+		 * For other cases, host class is set to NULL.
+		 */
+		if ((J2SE_VERSION(vm) >= J2SE_19) && (NULL != hostClass)) {
+			J9ROMClass *hostROMClass = hostClass->romClass;
+			/* This error-check should only be done for anonymous classes. */
+			Trc_BCU_Assert_True(isAnonFlagSet);
+			/* From Java 9 and onwards, set IllegalArgumentException when host class and anonymous class have different packages. */
+			if (!hasSamePackageName(romClass, hostROMClass)) {
+				PORT_ACCESS_FROM_JAVAVM(vm);
+				setIllegalArgumentExceptionHostClassAnonClassHaveDifferentPackages(vmThread, romClass, hostROMClass);
+				j9mem_free_memory(romClass);
+				omrthread_monitor_exit(vm->classTableMutex);
```

Comment:
It seems there is a mismatch for this `omrthread_monitor_exit(vm->classTableMutex)`.

Reply:
sender must hold class table mutex before calling `internalDefineClass`. class table mutex is released by `internalCreateRAMClassFromROMClass`. in case of error, we must release class table mutex before exiting `internalDefineClass` since error-check is done before the call to `internalCreateRAMClassFromROMClass`; otherwise, JVM will hang later ... during shutdown.